### PR TITLE
[portconfig]: Remove try block for db config initialization

### PR DESCRIPTION
**Pull request by SuvarnaMeenakshi**
#### Why I did it
1. Provide fix for comment: https://github.com/Azure/sonic-buildimage/pull/10475/files#r847753187; 
Follow up issue to be done in a different PR:
 https://github.com/Azure/sonic-buildimage/pull/10362/ - modified 2 test cases where the test case is required to get PORT table from config_db, the PR modified to use port_config.ini instead. 
#### How I did it
1. Try exception is not required in this scenario, so remove and modify to initial db config according to single or multi-asic platforms.

#### How to verify it
Verified on multi-asic device.
#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111

### CODE REVIEW

File: src/sonic-config-engine/portconfig.py [hunk 1]
```diff
@@ -72,12 +73,12 @@ def db_connect_configdb(namespace=None):
     """
     Connect to configdb
     """
-    try:
-        if namespace is not None:
+    if not swsscommon.SonicDBConfig.isInit():
+        if multi_asic.is_multi_asic():
             swsscommon.SonicDBConfig.load_sonic_global_db_config(namespace=namespace)
```
Comment (qiluo-msft) [inline src/sonic-config-engine/portconfig.py:78]: >load_sonic_global_db_config [](http://example.com/codeflow?start=37&length=27)

portconfig.py is just a library. It's the application's responsibility to load db config. Applications include:
1. sonic-cfggen
2. config
3. etc

File: src/sonic-config-engine/portconfig.py [hunk 2]
```diff
@@ -72,12 +73,12 @@ def db_connect_configdb(namespace=None):
...
-        config_db = swsscommon.ConfigDBConnector(use_unix_socket_path=True, namespace=namespace)
-    except Exception as e:
```
Comment (qiluo-msft) [inline src/sonic-config-engine/portconfig.py:79]: >Exception [](http://example.com/codeflow?start=11&length=9)

Could you use this PR for first issue in your description? And separate the second issue into another PR. The 2nd one is still under discussion.

File: src/sonic-config-engine/portconfig.py [hunk 3]
```diff
@@ -8,6 +8,7 @@
     from swsscommon import swsscommon
     from sonic_py_common import device_info
     from sonic_py_common.multi_asic import get_asic_id_from_name
+    from sonic_py_common import multi_asic
```
Comment (qiluo-msft) [inline src/sonic-config-engine/portconfig.py:11]: >multi_asic [](http://example.com/codeflow?start=32&length=10)

This line may trigger lgtm alert.

Comment (SuvarnaMeenakshi) [reply src/sonic-config-engine/portconfig.py]: Fixed lgtm alert

**Comment by jerseyang** [comment]
@[SuvarnaMeenakshi](https://github.com/SuvarnaMeenakshi), seems the new merge results my platform firsttime boot failed:

[   10.368939] rc.local[477]: + touch /tmp/pending_config_initialization
[   10.516144] rc.local[477]: + touch /tmp/notify_firstboot_to_platform
[   10.604155] rc.local[477]: + [ ! -d /host/reboot-cause/platform ]
[   10.680062] rc.local[477]: + [ -d /host/image-Add_Belgite_support.0-273d24a2a/platform/x86_64-cel_belgite-r0 ]
[   10.804153] rc.local[477]: + dpkg -i /host/image-Add_Belgite_support.0-273d24a2a/platform/x86_64-cel_belgite-r0/platform-modules-belgite_0.9_amd64.deb
[   10.975158] rc.local[615]: Selecting previously unselected package platform-modules-belgite.
[   11.088101] rc.local[615]: (Reading database ... 33373 files and directories currently installed.)
[   11.200160] rc.local[615]: Preparing to unpack .../platform-modules-belgite_0.9_amd64.deb ...
[   11.312187] rc.local[615]: Unpacking platform-modules-belgite (0.9) ...
[   11.404149] rc.local[615]: Setting up platform-modules-belgite (0.9) ...
[   11.494446] rc.local[2174]: Traceback (most recent call last):
[   11.572148] rc.local[2174]:   File "/usr/local/bin/sonic-cfggen", line 466, in <module>
[   11.672139] rc.local[2174]:     main()
[   11.720152] rc.local[2174]:   File "/usr/local/bin/sonic-cfggen", line 317, in main
[   11.824149] rc.local[2174]:     SonicDBConfig.load_sonic_db_config()
[   11.912181] rc.local[2174]:   File "/usr/lib/python3/dist-packages/swsscommon/swsscommon.py", line 1252, in load_sonic_db_config
[   12.056118] rc.local[2174]:     SonicDBConfig.initialize(sonic_db_file_path)
[   12.148125] rc.local[2174]:   File "/usr/lib/python3/dist-packages/swsscommon/swsscommon.py", line 1247, in initialize
[   12.280165] rc.local[2174]:     return _swsscommon.SonicDBConfig_initialize(*args, **kwargs)
[   12.392147] rc.local[2174]: RuntimeError: Sonic database config file doesn't exist at /var/run/redis/sonic-db/database_config.json
[   12.996604] rc.local[477]: + sync
[   13.213605] rc.local[477]: + [ -n x86_64-cel_belgite-r0 ]
[   13.284188] rc.local[477]: + [ -n  ]
[   13.336108] rc.local[477]: + mkdir -p /var/platform
[   13.400201] rc.local[477]: + [ -f /etc/default/kdump-tools ]
[   13.476168] rc.local[477]: + sed -i -e s/__PLATFORM__/x86_64-cel_belgite-r0/g /etc/default/kdump-tools
[   13.596152] rc.local[477]: + firsttime_exit
[   13.648166] rc.local[477]: + rm -rf /host/image-Add_Belgite_support.0-273d24a2a/platform/firsttime
[   13.768173] rc.local[477]: + exit 0
[   18.225351] DMAR: DRHD: handling fault status reg 3
[   18.285384] DMAR: [DMA Write] Request device [00:1a.0] PASID ffffffff fault addr 0 [fault reason 05] PTE Write access is not set

Debian GNU/Linux 11 sonic ttyS0

can you help to have a look? Thanks!

**Comment by qiluo-msft** [comment]
This commit could not be cleanly cherry-picked to 202012. Please submit another PR.

**Comment by SuvarnaMeenakshi** [comment]
> @[SuvarnaMeenakshi](https://github.com/SuvarnaMeenakshi), seems the new merge results my platform firsttime boot failed:
> 
> [ 10.368939] rc.local[477]: + touch /tmp/pending_config_initialization [ 10.516144] rc.local[477]: + touch /tmp/notify_firstboot_to_platform [ 10.604155] rc.local[477]: + [ ! -d /host/reboot-cause/platform ] [ 10.680062] rc.local[477]: + [ -d /host/image-Add_Belgite_support.0-273d24a2a/platform/x86_64-cel_belgite-r0 ] [ 10.804153] rc.local[477]: + dpkg -i /host/image-Add_Belgite_support.0-273d24a2a/platform/x86_64-cel_belgite-r0/platform-modules-belgite_0.9_amd64.deb [ 10.975158] rc.local[615]: Selecting previously unselected package platform-modules-belgite. [ 11.088101] rc.local[615]: (Reading database ... 33373 files and directories currently installed.) [ 11.200160] rc.local[615]: Preparing to unpack .../platform-modules-belgite_0.9_amd64.deb ... [ 11.312187] rc.local[615]: Unpacking platform-modules-belgite (0.9) ... [ 11.404149] rc.local[615]: Setting up platform-modules-belgite (0.9) ... [ 11.494446] rc.local[2174]: Traceback (most recent call last): [ 11.572148] rc.local[2174]: File "/usr/local/bin/sonic-cfggen", line 466, in [ 11.672139] rc.local[2174]: main() [ 11.720152] rc.local[2174]: File "/usr/local/bin/sonic-cfggen", line 317, in main [ 11.824149] rc.local[2174]: SonicDBConfig.load_sonic_db_config() [ 11.912181] rc.local[2174]: File "/usr/lib/python3/dist-packages/swsscommon/swsscommon.py", line 1252, in load_sonic_db_config [ 12.056118] rc.local[2174]: SonicDBConfig.initialize(sonic_db_file_path) [ 12.148125] rc.local[2174]: File "/usr/lib/python3/dist-packages/swsscommon/swsscommon.py", line 1247, in initialize [ 12.280165] rc.local[2174]: return _swsscommon.SonicDBConfig_initialize(*args, **kwargs) [ 12.392147] rc.local[2174]: RuntimeError: Sonic database config file doesn't exist at /var/run/redis/sonic-db/database_config.json [ 12.996604] rc.local[477]: + sync [ 13.213605] rc.local[477]: + [ -n x86_64-cel_belgite-r0 ] [ 13.284188] rc.local[477]: + [ -n ] [ 13.336108] rc.local[477]: + mkdir -p /var/platform [ 13.400201] rc.local[477]: + [ -f /etc/default/kdump-tools ] [ 13.476168] rc.local[477]: + sed -i -e s/**PLATFORM**/x86_64-cel_belgite-r0/g /etc/default/kdump-tools [ 13.596152] rc.local[477]: + firsttime_exit [ 13.648166] rc.local[477]: + rm -rf /host/image-Add_Belgite_support.0-273d24a2a/platform/firsttime [ 13.768173] rc.local[477]: + exit 0 [ 18.225351] DMAR: DRHD: handling fault status reg 3 [ 18.285384] DMAR: [DMA Write] Request device [00:1a.0] PASID ffffffff fault addr 0 [fault reason 05] PTE Write access is not set
> 
> Debian GNU/Linux 11 sonic ttyS0
> 
> can you help to have a look? Thanks!

Thanks for reporting, I will  check this error , will provide a clean fix.

**Comment by qiluo-msft** [comment]
Since this commit is later merged on master. I will not cherry-pick to 202012.
